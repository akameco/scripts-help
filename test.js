// @flow
const fs = require('fs')
const { help, create } = require('.')

const spyLog = jest.spyOn(console, 'log').mockImplementation(v => v)
const writeSpy = jest.spyOn(fs, 'writeFileSync').mockImplementation(v => v)

test('help snapshot', async () => {
  await help()
  expect(spyLog.mock.calls[0][0]).toMatchSnapshot()
})

test('create snapshot', async () => {
  await create()
  expect(writeSpy.mock.calls[0][1]).toMatchSnapshot()
})
