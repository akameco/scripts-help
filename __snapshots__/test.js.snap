// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot 1`] = `
"{
  \\"add-contributor\\": \\"\\",
  \\"precommit\\": \\"\\",
  \\"fmt\\": \\"\\",
  \\"flow\\": \\"\\",
  \\"lint\\": \\"\\",
  \\"test\\": \\"\\",
  \\"test:ci\\": \\"\\",
  \\"help\\": \\"\\"
}"
`;

exports[`help snapshot 1`] = `
"  [1madd-contributor[22m
[2m      \`add-contributor ユーザ名 code\`でreadmeにコントリビューターとして追加します[22m

  [1mprecommit[22m
[2m      No description[22m

  [1mfmt[22m
[2m      prettierのバージョンを上げたときに実行[22m

  [1mflow[22m
[2m      flowを実行[22m

  [1mlint[22m
[2m      eslintを実行[22m

  [1mtest[22m
[2m      テストを実行[22m

  [1mtest:cov[22m
[2m      カバレッジありでjestを実行[22m

  [1mtest:ci[22m
[2m      No description[22m

  [1mhelp[22m
[2m      helpを表示[22m
"
`;
